tests/cli/archive/test_archive_cli.py

```diff
@@ -0,0 +1,26 @@
+from click.testing import CliRunner
+
+from cg.cli.archive import archive_spring_files
+from cg.models.cg_config import CGConfig
+
+
+def test_limit_and_archive_all_fails(cli_runner: CliRunner, cg_context: CGConfig):
+    """Tests that when invoking archive-spring-files in the Archive CLI module, the command is aborted
+    if both a limit and the --archive-all flag is provided."""
+
+    # GIVEN a CLI runner and a context
+
+    # WHEN invoking archive_spring_files with both a given limit and specifying archive_all
+    result = cli_runner.invoke(
+        archive_spring_files,
+        ["--limit", 100, "--archive-all"],
+        obj=cg_context,
+    )
+
+    # THEN the command should have exited with an exit_code 1
+    assert result.exit_code == 1
+    assert isinstance(result.exception, SystemExit)
+    assert (
+        "Incorrect input parameters - please do not provide both a limit and set --archive-all."
+        in result.stdout
+    )
```

Comment:
UGLY but I have not found a way to test if the behaviour is correct in a better way

Reply:
why not don't you are an input error or something that can be checked for here?

Reply:
So Click's CliRunner seems to be able to forward any Exception apart from Click's (e.g. Abort). But changing from click.Abort would probably result in a printing of the stacktrace, which I think would be worse than having a test which is ugly.